### PR TITLE
feat(grouping): Speed up similarity backfilling

### PR DESCRIPTION
**Pull request by armenzg**
This will speed up the number of tasks that get scheduled.

### CODE REVIEW

File: src/sentry/tasks/embeddings_grouping/utils.py [hunk 1]
```diff
@@ -110,9 +116,11 @@ def create_project_cohort(worker_number: int, last_processed_project_id: int | N
     total_worker_count = options.get("similarity.backfill_total_worker_count")
     cohort_size = options.get("similarity.backfill_project_cohort_size")
 
+    query = Project.objects.filter(project_id_filter)
+    if skip_processed_projects:
+        query = query.exclude(projectoption__key=PROJECT_BACKFILL_COMPLETED)
```
Comment (jangjodi) [inline src/sentry/tasks/embeddings_grouping/utils.py:121]: I think we can remove these [lines](https://github.com/getsentry/sentry/blob/abe3cd7014a8f81cae6ad29c76501ac14106012b/src/sentry/tasks/embeddings_grouping/backfill_seer_grouping_records_for_project.py#L138-L151) as it'll be redundant to this change

Comment (armenzg) [reply src/sentry/tasks/embeddings_grouping/utils.py:122]: I have added the tests. I will look into this comment now.

File: tests/sentry/tasks/test_backfill_seer_grouping_records.py [hunk 2]
```diff
@@ -1763,53 +1766,73 @@ def test_backfill_seer_grouping_records_skip_project_already_processed(self, moc
         assert mock_logger.info.call_args_list == expected_call_args_list
 
     @with_feature("projects:similarity-embeddings-backfill")
+    @override_options({"similarity.backfill_total_worker_count": 1})
+    @override_options({"similarity.new_project_seer_grouping.enabled": True})
     @patch("sentry.tasks.embeddings_grouping.backfill_seer_grouping_records_for_project.logger")
     @patch("sentry.tasks.embeddings_grouping.utils.post_bulk_grouping_records")
-    def test_backfill_seer_grouping_records_reprocess_project_already_processed(
+    def test_backfill_seer_grouping_records_typical_backfill_request(
```
Comment (armenzg) [inline tests/sentry/tasks/test_backfill_seer_grouping_records.py:1773]: I'm changing this test to represent for the most normal backfill request with workers.

File: tests/sentry/tasks/test_backfill_seer_grouping_records.py [hunk 3]
```diff
@@ -1754,53 +1764,73 @@ def test_backfill_seer_grouping_records_skip_project_already_processed(self, moc
...
+                last_processed_group_id_input=None,
+                skip_processed_projects=True,
+                enable_ingestion=True,
+                worker_number=0,
+            )
 
+        key = "backfill_seer_grouping_records"
+        # Since we set the total worker count to 1, the project cohort will have all projects
+        # except the one that has already been backfilled
+        cohort = [self.project.id, project3.id]
```
Comment (jangjodi) [inline tests/sentry/tasks/test_backfill_seer_grouping_records.py:1797]: if we want to test that `skip_processed_projects=True` works as intended, shouldn't we include `project2`, the project that is already processed, in the cohort and have the test check that it's not being re-processed?

Comment (jangjodi) [reply tests/sentry/tasks/test_backfill_seer_grouping_records.py:1796]: thanks for following up on slack!

File: tests/sentry/tasks/test_backfill_seer_grouping_records.py [hunk 4]
```diff
@@ -1763,53 +1763,73 @@ def test_backfill_seer_grouping_records_skip_project_already_processed(self, moc
...
+        # Create two more projects and one of them is already backfilled
+        project2 = self.create_project(organization=self.organization)
+        project2.update_option(PROJECT_BACKFILL_COMPLETED, int(time.time()))
+        project3 = self.create_project(organization=self.organization)
+
         mock_post_bulk_grouping_records.return_value = {"success": True, "groups_with_neighbor": {}}
-        self.project.update_option(PROJECT_BACKFILL_COMPLETED, int(time.time()))
         with TaskRunner():
-            backfill_seer_grouping_records_for_project(self.project.id, None)
+            # This is the typical way of backfilling all projects
```
Comment (armenzg) [inline tests/sentry/tasks/test_backfill_seer_grouping_records.py:1784]: This is representing the backfill call: `{"enable_ingestion":"foo","worker_number":0,"skip_processed_projects":"bar"}`

**Review by jangjodi — APPROVED** [review]

**Pull request merged**